### PR TITLE
implement bounds check

### PR DESCRIPTION
**Pull request by Johann150**
This adds functionality to check whether any coordinates of a job lie outside of the workspace. Only things that are selected for a pass are taken into account and the current offset is respected to cover even theoretical/esoteric use cases, some of which I came across when talking with other lab members, for example:
- not having to remove a path that is outside of the work area if you can just paint it in a different colour
- drawing a circle around the origin of the SVG and using the offset to determine where the midpoint of the circle is

To help users understand the error, I've also increased the time error messages are displayed from 3s to 6s to give people more time to read. I tried making them "sticky" so you have to actively dismiss them, but had some trouble doing that.

I also did some miscellaneous code clean-up and simplifications.

### CODE REVIEW

File: backend/driveboard.py [hunk 1]
```diff
@@ -1103,6 +1070,74 @@ def job(jobdict):
...
+
+    with SerialLoop.lock:
+        x_off = SerialLoop._status['offset'][0]
+        y_off = SerialLoop._status['offset'][1]
+    x_lim = conf['workspace'][0] - x_off
+    y_lim = conf['workspace'][1] - y_off
+
+    # loop passes
+    for passidx, pass_ in enumerate(jobdict['passes']):
+        def check_point(point, kind):
```
Comment (scottshambaugh) [inline backend/driveboard.py]: Could you move this definition out of the for loop? Otherwise, these changes all look great!

Comment (Johann150) [reply backend/driveboard.py]: `check_point` uses/depends on `passidx`. I'm not too familiar with scoping around function definitions and closures in python with this, so I'm not sure if moving the definition outside of the loop would work?

Comment (scottshambaugh) [reply backend/driveboard.py]: Could you pass in `passidx` as an argument? I think it should work fine as written, but to your point it's messy from a scoping perspective.

**Comment by scottshambaugh** [comment]
Looks good now, thanks!

**Pull request merged**